src/formatDate.js

```diff
@@ -7,8 +7,58 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeperator = fromFormat[fromFormat.length - 1];
+  const dateParts = date.split(fromSeperator);
+  const indexes = {
+    YYYY: fromFormat.indexOf('YYYY'),
+    YY: fromFormat.indexOf('YY'),
+    MM: fromFormat.indexOf('MM'),
+    DD: fromFormat.indexOf('DD'),
+  };
+
+  const newDate = [];
+
+  for (let i = 0; i < toFormat.length; i++) {
+    const part = toFormat[i];
+
+    if (part !== toFormat[toFormat.length - 1]) {
+      if (part === 'YYYY') {
+        if (dateParts[indexes['YYYY']] !== undefined) {
+          newDate.push(dateParts[indexes['YYYY']]);
+        } else {
+          newDate.push(convertYear(dateParts[indexes['YY']], 'YY'));
+        }
+      } else if (part === 'YY') {
+        if (dateParts[indexes['YY']] !== undefined) {
+          newDate.push(dateParts[indexes['YY']]);
+        } else {
+          newDate.push(dateParts[indexes['YYYY']].slice(-2));
+        }
+      } else {
+        newDate.push(dateParts[indexes[part]]);
+      }
+    }
```

Comment:
The loop for constructing the new date can be simplified by moving the condition for the separator outside of the loop. This would prevent checking whether each part is a separator on every iteration, which is unnecessary since the separator is always at the end of the format array.